lib/module-resolvers/file-resolver.js

```diff
@@ -36,6 +55,23 @@ class FileResolver {
         }
     }
 
+    _read_file(filename, name) {
+
+        if (! this._is_file(filename)) {
+            return false;
+        }
+
+        try {
+            return {
+                source: fs.readFileSync(filename, 'utf8'),
+                name: (name ? name : filename)
+            };
+        }
+        catch (e) {
+            return false;
+        }
+    }
+
```

Comment:
The `_read_file` extraction would be better in a separate commit (it is orthogonal with the other changes).